sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/EntityDataModelType.java

```diff
@@ -0,0 +1,151 @@
+
+    /**
+     * <strong>Edm.Decimal</strong> Represents numeric values with fixed precision and scale. This type can describe a
+     * numeric value ranging from negative 10^255 + 1 to positive 10^255 -1
+     */
+    DECIMAL("Edm.Decimal", Collections.singleton(BigDecimal.class), BigDecimal::new),
+
+    /**
+     * <strong>Edm.Double</strong> Represents a floating point number with 15 digits precision that can represent values
+     * with approximate range of +/- 2.23e -308 through +/- 1.79e +308
+     */
+    DOUBLE("Edm.Double", Stream.of(double.class, Double.class), Double::parseDouble),
+
+    /**
+     * <strong>Edm.Single</strong> Represents a floating point number with 7 digits precision that can represent values
+     * with approximate range of +/- 1.18e -38 through +/- 3.40e +38
+     */
+    SINGLE("Edm.Single", Stream.of(float.class, Float.class), Float::parseFloat),
+
+    /**
+     * <strong>Edm.Guid</strong> Represents a 16-byte (128-bit) unique identifier value
+     */
+    GUID("Edm.Guid", Collections.singleton(UUID.class), UUID::fromString),
+
+    /**
+     * <strong>Edm.Int16</strong> Represents a signed 16-bit integer value
+     */
+    INT16("Edm.Int16", Stream.of(short.class, Short.class), Short::parseShort),
+
+    /**
+     * <strong>Edm.Int32</strong> Represents a signed 32-bit integer value
+     */
+    INT32("Edm.Int32", Stream.of(int.class, Integer.class), Integer::parseInt),
+
+    /**
+     * <strong>Edm.Int64</strong> Represents a signed 64-bit integer value
+     */
+    INT64("Edm.Int64", Stream.of(long.class, Long.class), Long::parseLong),
+
+    /**
+     * <strong>Edm.SByte</strong> Represents a signed 8-bit integer value
+     */
+    SBYTE("Edm.SByte", Stream.of(byte.class, Byte.class), Byte::parseByte),
+
+    /**
+     * <strong>Edm.String</strong> Represents fixed- or variable-length character data
+     */
+    STRING("Edm.String", Collections.singleton(String.class), s -> s),
+
+    /**
+     * <strong>Edm.Time</strong> Represents the time of day with values ranging from 0:00:00.x to 23:59:59.y, where x
+     * and y depend upon the precision
+     */
+    TIME("Edm.Time", Stream.of(LocalTime.class, OffsetTime.class), s ->
+        DateTimeFormatter.ISO_DATE_TIME.parseBest(s, OffsetTime::from, LocalTime::from)
+    );
+
+    private final String edmType;
+    private final Set<Class<?>> serializeFrom;
+    private final Function<String, Object> deserializer;
+
+    EntityDataModelType(String edmType, Set<Class<?>> serializeFrom, Function<String, Object> deserializer) {
+        this.edmType = edmType;
+        this.serializeFrom = serializeFrom;
+        this.deserializer = deserializer;
+    }
+
+    EntityDataModelType(String edmType, Stream<Class<?>> serializeFrom, Function<String, Object> deserializer) {
+        this(edmType, (Set<Class<?>>) serializeFrom.collect(Collectors.toCollection(HashSet::new)), deserializer);
+    }
+
+    public static EntityDataModelType parse(String edmType) {
```

Comment:
I think the convention is valueOf

Reply:
There is already a `valueOf(String)` method which is provided automatically for all Enums, but it takes the string value of the enum literal (and loads it by reflection). E.g. "BINARY". I would be open to other names for this method though.

Reply:
Resolved by changing to `fromString` which matches what Autorest does.